libs/core/build.gradle

```diff
@@ -117,3 +117,8 @@ tasks.named("dependencyLicenses").configure {
   mapping from: /jackson-.*/, to: 'jackson'
   mapping from: /lucene-.*/, to: 'lucene'
 }
+
+tasks.withType(JavaCompile).configureEach {
+  options.compilerArgs -= '-Xlint:rawtypes'
+  options.compilerArgs -= '-Xlint:unchecked'
+}
```

Comment:
What required these?

Reply:
https://bugs.openjdk.org/browse/JDK-6747671 - warning on raw types, certainly good to have :-)

Reply:
Oh it's `-=` those 🤔

Reply:
Oh you are right - we should not do that but cleanup the code

Reply:
[Rote](https://github.com/opensearch-project/OpenSearch/blob/131a341e0dd16e642d6e424ec6460746a93cf340/server/build.gradle#L161).  It's a non issue a needed for generics. Which is why it's used in most modules.

Reply:
> [Rote](https://github.com/opensearch-project/OpenSearch/blob/131a341e0dd16e642d6e424ec6460746a93cf340/server/build.gradle#L161). It's a non issue and needed for generics. Which is why it's used in most modules.

This is one of these setting that we should not replicate but reduce: there are limited places when raw types / unchecked warning suppression may be needed but for majority of the code this should not generate any warning, especially the new code we are adding.